src/components/Messages/index.tsx

```diff
@@ -29,67 +69,63 @@ const Messages: FC = () => {
         setClient(xmtp);
       }
     };
-    if (isFeatureEnabled('messages', currentProfile?.id)) {
+    if (isMessagesEnabled) {
       initXmtpClient();
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [signer]);
 
   useEffect(() => {
-    if (!client || !isFeatureEnabled('messages', currentProfile?.id)) {
+    if (!isMessagesEnabled || !client) {
       return;
     }
 
-    async function listConversations() {
-      setLoading(true);
+    const fetchMostRecentMessage = async (
+      convo: Conversation
+    ): Promise<{ address: string; preview: MessagePreview }> => {
+      const peerAddress = convo.peerAddress.toLowerCase();
+      const newMessagePreview = new MessagePreview(messagePreviews.get(peerAddress));
+      // TODO(elise): Add sort direction on XMTP's side so we can grab only the most recent message.
+      const newMessages = await convo.messages({ limit: 1 });
+      if (newMessages.length >= 0) {
+        newMessagePreview.message = newMessages[0];
+      }
+      return { address: peerAddress, preview: newMessagePreview };
+    };
+
+    const listConversations = async () => {
+      const newMessagePreviews = new Map(messagePreviews);
+      const newConversations = new Map(conversations);
       const convos = (await client?.conversations?.list()) || [];
-      Promise.all(
+      const previews = await Promise.all(
         convos.map(async (convo) => {
-          if (convo.peerAddress !== currentProfile?.ownedBy) {
-            conversations.set(convo.peerAddress, convo);
-            setConversations(new Map(conversations));
-          }
+          newConversations.set(convo.peerAddress, convo);
+          return await fetchMostRecentMessage(convo);
         })
-      ).then(() => {
-        setLoading(false);
-      });
-    }
-    const streamConversations = async () => {
```

Comment:
Any reason we removed the streaming of conversations?

We need this when new conversation is started with you, that is already not listed

Reply:
Yeah, I should've added this to the PR description, but I removed it for simplicity for now because I wanted to nail down the mapping of the paginated `Profiles` query to the non-paginated convos/messages fetch first. Didn't want another hook potentially updating the state _yet_, but I opened an issue to add this back in a follow-up here & assigned to myself: https://github.com/xmtp-labs/hq/issues/732.